src/core/filters/meta/filter.rs

```diff
@@ -0,0 +1,467 @@
+//! # FilterMeta
+//!
+//! Object for packet filtering it implements from_string(). It takes
+//! as input a filter string under the form
+//! struct_name>member1>member2>..>leafmember generating a sequence of
+//! actions.
+
+use anyhow::{anyhow, bail, Result};
+use btf_rs::*;
+use plain::Plain;
+use regex::Regex;
+
+use crate::core::inspect::inspector;
+
+const META_OPS_MAX: u32 = 32;
+const META_TARGET_MAX: usize = 32;
+const MF_RE: &str = r#"([a-zA-Z_][a-zA-Z0-9_]*(?:>[a-zA-Z_][a-zA-Z0-9_]*){1,})[\s]{1,}(==|!=|>=|<=|<|>)[\s]{1,}(0x[0-9a-fA-F]+|-?[1-9][0-9]*|"[ -~]+"|[a-zA-Z_][a-zA-Z0-9_]*){1}"#;
```

Comment:
ok, the zero problem is because of an unneeded last-minute change (avoid 0123 as, based on the logic, they represent octal nums).
Of course, we can tokenize and parse in different ways, but that seemed handy (and it was).
If we have a preference for the non-regex approach, sgtm.

Reply:
my point was just regexps are largely used for tokens (of course in a slightly different way).
There are ways to simplify (e.g. we can split in multiple static strings).
If this is so important, we can do it manually.